base/replutil.jl

```diff
@@ -196,10 +196,32 @@ function showerror(io::IO, ex::TypeError)
 end
 
 function showerror(io::IO, ex, bt; backtrace=true)
-    try
-        showerror(io, ex)
-    finally
-        backtrace && show_backtrace(io, bt)
+    if get(io, :REPLError, false)
+        if backtrace
+            io_bt = IOBuffer()
+            io_bt_con = IOContext(io_bt, io)
+            show_backtrace(io_bt_con, bt)
+            backtrace_str = takebuf_string(io_bt)
+            # Only print the backtrace header if there actually is a printed backtrace
+            if backtrace_str != ""
+                header = string(typeof(ex).name.name)
+                line_len = 76
```

Comment:
> no matter the size of the terminal...

Maybe due to the `Base.with_output_color`? Doesn't it use a separate buffer to print the colours?